operator/v1/types_machineconfiguration.go

```diff
@@ -56,3 +65,98 @@ type MachineConfigurationList struct {
 	// Items contains the items
 	Items []MachineConfiguration `json:"items"`
 }
+
+type ManagedBootImages struct {
+	// machineManagers is an array of machineManager objects.
+	// The MCO will watch for changes to this list and register/de-register machine management resources from boot image updates.
+	// An entry in this list consists of the resource type, the API group that the resource belongs to and a selection filter
+	// on the resources.
+	//
+	// Warning: Only one entry is permitted per unique pair of resource/API group. The label selector provided within MachineManager
+	// can be used for further customization if required.
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=resource
+	// +listMapKey=apiGroup
+	MachineManagers []MachineManager `json:"machineManagers"`
+}
+
+// MachineManager contains identifying information of a machine management resource(eg. a machineset) that will be
+// registered for boot image updates. This is likely to evolve as support for more machine management resources are added.
+type MachineManager struct {
+	// resource is the machine management resource's type.
+	//
+	// The following values are accepted:
+	// - MachineSets: The machine manager will only register resources of the type MachineSet, which may belong to MachineAPI or ClusterAPI.
+	//
+	// +kubebuilder:validation:Required
+	Resource MachineManagerMachineSetsResourceType `json:"resource"`
+	// apiGroup is name of the APIGroup that the machine management resource belongs to.
+	//
+	// The following values are accepted:
+	// - MachineAPI: The machine manager will only register resources that belong to MachineAPI APIGroup.
+	//
+	// +kubebuilder:validation:Required
+	APIGroup MachineManagerMachineSetsAPIGroupType `json:"apiGroup"`
+	// selection allows granular control of the machine management resources that will be registered for boot image updates.
+	//
+	// +kubebuilder:validation:Required
+	Selection MachineManagerSelector `json:"selection"`
+}
+
+// +kubebuilder:validation:XValidation:rule="has(self.mode) && self.mode == 'Partial' ?  has(self.partial) : !has(self.partial)",message="Partial is required when type is partial, and forbidden otherwise"
+// +union
+type MachineManagerSelector struct {
+	// mode is a union discriminator for MachineManagerSelector and can have three possible values.
+	// - All: All resources specified by the parent MachineManager are registered for boot image updates.
+	// - None: No resources specified by the parent MachineManager are registered for boot image updates.
+	// - Partial: resources specified by the parent MachineManager are registered for boot image updates only if they match with the label selector.
```

Comment:
Have you tried an `oc explain` of these fields with the current patch? I'm not sure the formatting will carry over and it would be good to check how this looks in an `oc explain`

Reply:
oof, yeah I missed that. Just tried it, it's not pretty. Will reformat.